testing/tests/consensus_task.rs

```diff
@@ -1,346 +1,344 @@
+            panic!("Failed to find genesis leaf.");
+        };
+        leaf.clone()
+    } else {
+        consensus
+            .saved_leaves
+            .get(&justify_qc.leaf_commitment())
+            .cloned()
+            .unwrap()
+    };
+
+    let parent_commitment = parent.commit();
+
+    let leaf: SequencingLeaf<_> = SequencingLeaf {
+        view_number: view,
+        height: proposal.height,
+        justify_qc: proposal.justify_qc.clone(),
+        parent_commitment,
+        deltas: Right(proposal.block_commitment),
+        rejected: Vec::new(),
+        timestamp: 0,
+        proposer_id: quorum_exchange.get_leader(view).to_bytes(),
+    };
+
+    quorum_exchange.create_yes_message(
+        proposal.justify_qc.commit(),
+        leaf.commit(),
+        view,
+        vote_token,
+    )
+}
+
+#[cfg(test)]
+#[cfg_attr(
+    feature = "tokio-executor",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(feature = "async-std-executor", async_std::test)]
+async fn test_consensus_task() {
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+
+    let handle = build_consensus_api(1).await;
+    let (private_key, public_key) = key_pair_for_id(1);
+
+    let mut input = Vec::new();
+    let mut output = HashMap::new();
+
+    input.push(SequencingHotShotEvent::ViewChange(ViewNumber::new(1)));
+    input.push(SequencingHotShotEvent::ViewChange(ViewNumber::new(2)));
+    input.push(SequencingHotShotEvent::Shutdown);
+
+    output.insert(
+        build_proposal_send(&handle, &private_key, &public_key).await,
+        1,
+    );
+    output.insert(SequencingHotShotEvent::ViewChange(ViewNumber::new(1)), 2);
+    output.insert(SequencingHotShotEvent::ViewChange(ViewNumber::new(2)), 2);
+    output.insert(SequencingHotShotEvent::Shutdown, 1);
+
+    let build_fn = |task_runner, event_stream| {
+        add_consensus_task(task_runner, event_stream, ChannelStream::new(), handle)
+    };
+
+    run_harness(input, output, build_fn).await;
+}
+
+#[cfg(test)]
+#[cfg_attr(
+    feature = "tokio-executor",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(feature = "async-std-executor", async_std::test)]
+async fn test_consensus_vote() {
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+
+    let handle = build_consensus_api(2).await;
+    let (private_key, public_key) = key_pair_for_id(1);
+
+    let mut input = Vec::new();
+    let mut output = HashMap::new();
+
+    let propsal_recv = build_proposal_recv(&handle, &private_key, &public_key).await;
+
+    input.push(SequencingHotShotEvent::ViewChange(ViewNumber::new(1)));
+    input.push(propsal_recv.clone());
+
+    input.push(SequencingHotShotEvent::Shutdown);
+
+    output.insert(propsal_recv.clone(), 1);
+    if let SequencingHotShotEvent::QuorumProposalRecv(proposal, _) = propsal_recv {
```

Comment:
This `if` seems to be redundant since `proposal_recv` is always `SequencingHotShotEvent::QuorumProposalRecv`. Would it be better if we just make the `build_proposal_recv` return `(message, key)` rather than the `SequencingHotShotEvent` enum, then use it to build the enum to be pushed to `input` in this function?

However, except for the additional `if` check, I do think the way you implement looks nicer, so I don't have a strong opinion here. 😃